ci-operator/step-registry/ipi/deprovision/deprovision/in-bastion/ipi-deprovision-deprovision-in-bastion-commands.sh

```diff
@@ -60,24 +73,71 @@ BASTION_SSH_USER=$(<"${SHARED_DIR}/bastion_ssh_user")
+        AZURE_AUTH_LOCATION="${SHARED_DIR}/azure_managed_identity_osServicePrincipal.json"
+    else
+        AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+    fi
+    run_scp_to_remote "${SSH_PRIV_KEY_PATH}" "${BASTION_SSH_USER}" "${BASTION_IP}" "${AZURE_AUTH_LOCATION}" "${REMOTE_DIR}/osServicePrincipal.json"
+    echo "export AZURE_AUTH_LOCATION='${REMOTE_DIR}/osServicePrincipal.json'" >> "${REMOTE_ENV_FILE}"
+    ;;
+gcp)
+    if [[ -z "${ATTACH_BASTION_SA}" ]]; then
+        echo "$(date -u --rfc-3339=seconds) - Uploading GCP credential to bastion"
+        GOOGLE_CLOUD_KEYFILE_JSON=${CLUSTER_PROFILE_DIR}/gce.json
+        run_scp_to_remote "${SSH_PRIV_KEY_PATH}" "${BASTION_SSH_USER}" "${BASTION_IP}" "${GOOGLE_CLOUD_KEYFILE_JSON}" "${REMOTE_DIR}/gce.json"
+        echo "export GOOGLE_CLOUD_KEYFILE_JSON='${REMOTE_DIR}/gce.json'" >> "${REMOTE_ENV_FILE}"
+    else
+        echo "$(date -u --rfc-3339=seconds) - The cluster destroying on bastion will use the service-account attached to the bastion host, nothing to do"
+    fi
+    ;;
+*) >&2 echo "$(date -u --rfc-3339=seconds) - No need to upload any credential files into bastion host for cluster type '${CLUSTER_TYPE}'"
+esac
+
+run_ssh_cmd "${SSH_PRIV_KEY_PATH}" "${BASTION_SSH_USER}" "${BASTION_IP}" "mkdir -p ${REMOTE_INSTALL_DIR}"
 
-echo "Deprovisioning cluster ..."
 if [[ ! -s "${SHARED_DIR}/metadata.json" ]]; then
-  echo "Skipping: ${SHARED_DIR}/metadata.json not found."
-  exit 0
+  echo "$(date -u --rfc-3339=seconds) - ERROR: ${SHARED_DIR}/metadata.json not found."
+  exit 1
 fi
 
-echo ${SHARED_DIR}/metadata.json
+echo "$(date -u --rfc-3339=seconds) - Uploading '${SHARED_DIR}/metadata.json' to bastion"
+run_scp_to_remote "${SSH_PRIV_KEY_PATH}" "${BASTION_SSH_USER}" "${BASTION_IP}" "${SHARED_DIR}/metadata.json" "${REMOTE_INSTALL_DIR}"
+
+INSTALLER_BINARY=$(which openshift-install)
+
+if [[ -n "${CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE:-}" ]]; then
+  CUSTOM_PAYLOAD_DIGEST=$(oc adm release info "${CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" -a "${CLUSTER_PROFILE_DIR}/pull-secret" --output=jsonpath="{.digest}")
+  CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="${CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE%:*}"@"$CUSTOM_PAYLOAD_DIGEST"
+  echo "$(date -u --rfc-3339=seconds) - Extracting installer from ${CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+  oc adm release extract -a "${CLUSTER_PROFILE_DIR}/pull-secret" "${CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" \
+  --command=openshift-install --to="/tmp" || exit 1
+  export INSTALLER_BINARY="/tmp/openshift-install"
+fi
+
+echo "$(date -u --rfc-3339=seconds) - Uploading '${INSTALLER_BINARY}' to bastion"
+run_scp_to_remote "${SSH_PRIV_KEY_PATH}" "${BASTION_SSH_USER}" "${BASTION_IP}" "${INSTALLER_BINARY}" "${REMOTE_DIR}"
+
+echo "$(date -u --rfc-3339=seconds) - Going to destroy cluster from release ${CUSTOM_OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
```

Comment:
The line 119 already indicated where the installer was extracted, so I do not think we need this line any more.